### PR TITLE
 Tests I

### PR DESCRIPTION
**Pull request by mflis**
First part of tests. More on the way

### CODE REVIEW

File: what_to_test.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+user
```
Comment (maxmati) [inline what_to_test.md]: Isn't it done?

**Comment by maxmati** [comment]
Any instructions how to run it?
I'm getting fallowing error:
```
maxmati@leopardus ~/s/2/m/z/zmora-server-rails (test ↩☡=) env RAILS_ENV=test bundler exec -- rake db:migrate
rake aborted!
LoadError: cannot load such file -- rails_helper
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:292:in `require'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:292:in `block in require'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:258:in `load_dependency'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:292:in `require'
/home/maxmati/src/2017/maxmati/zmora/zmora-server-rails/spec/factories/contest_factory.rb:1:in `<top (required)>'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:286:in `load'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:286:in `block in load'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:258:in `load_dependency'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:286:in `load'
/home/maxmati/.gem/ruby/2.2.0/gems/factory_girl-4.8.0/lib/factory_girl/find_definitions.rb:20:in `block (2 levels) in find_definitions'
/home/maxmati/.gem/ruby/2.2.0/gems/factory_girl-4.8.0/lib/factory_girl/find_definitions.rb:19:in `each'
/home/maxmati/.gem/ruby/2.2.0/gems/factory_girl-4.8.0/lib/factory_girl/find_definitions.rb:19:in `block in find_definitions'
/home/maxmati/.gem/ruby/2.2.0/gems/factory_girl-4.8.0/lib/factory_girl/find_definitions.rb:15:in `each'
/home/maxmati/.gem/ruby/2.2.0/gems/factory_girl-4.8.0/lib/factory_girl/find_definitions.rb:15:in `find_definitions'
/home/maxmati/.gem/ruby/2.2.0/gems/factory_girl_rails-4.8.0/lib/factory_girl_rails/railtie.rb:21:in `block in <class:Railtie>'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/lazy_load_hooks.rb:43:in `call'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/lazy_load_hooks.rb:43:in `execute_hook'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/lazy_load_hooks.rb:52:in `block in run_load_hooks'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/lazy_load_hooks.rb:51:in `each'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/lazy_load_hooks.rb:51:in `run_load_hooks'
/home/maxmati/.gem/ruby/2.2.0/gems/railties-5.1.1/lib/rails/application/finisher.rb:73:in `block in <module:Finisher>'
/home/maxmati/.gem/ruby/2.2.0/gems/railties-5.1.1/lib/rails/initializable.rb:30:in `instance_exec'
/home/maxmati/.gem/ruby/2.2.0/gems/railties-5.1.1/lib/rails/initializable.rb:30:in `run'
/home/maxmati/.gem/ruby/2.2.0/gems/railties-5.1.1/lib/rails/initializable.rb:59:in `block in run_initializers'
/home/maxmati/.gem/ruby/2.2.0/gems/railties-5.1.1/lib/rails/initializable.rb:58:in `run_initializers'
/home/maxmati/.gem/ruby/2.2.0/gems/railties-5.1.1/lib/rails/application.rb:353:in `initialize!'
/home/maxmati/src/2017/maxmati/zmora/zmora-server-rails/config/environment.rb:5:in `<top (required)>'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:292:in `require'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:292:in `block in require'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:258:in `load_dependency'
/home/maxmati/.gem/ruby/2.2.0/gems/activesupport-5.1.1/lib/active_support/dependencies.rb:292:in `require'
/home/maxmati/.gem/ruby/2.2.0/gems/railties-5.1.1/lib/rails/application.rb:329:in `require_environment!'
/home/maxmati/.gem/ruby/2.2.0/gems/railties-5.1.1/lib/rails/application.rb:445:in `block in run_tasks_blocks'
/home/maxmati/.gem/ruby/2.2.0/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
Tasks: TOP => db:migrate => environment
(See full trace by running task with --trace)
```

File: what_to_test.md (hunk 1 above)
Comment (mflis) [reply what_to_test.md]: forgot to delete this file

**Comment by mflis** [comment]
I use  `bin/bundle exec rspec`.
Or `bin/bundle exec guard --clear --notify false` to run in continuous mode


From what I observed rake has problems with  `require 'rails_helper'` on the beginning of each spec. 

It works well  with paths relative to project root e.g. `require './spec/rails_helper.rb'`, but after fixing these problems others come on the way so I gave up with rake

**Comment by maxmati** [comment]
but when i use `bin/bundle exec rspec` it asks to do db:migrate first

**Review by marcmos — APPROVED** [review]
@maxmati, jak nic nie masz, to merge'ujemy. Jest dużo commitów, może squash (mnie jest obojętne)?

**Pull request merged**